pkg/operator/baremetal_config.go

```diff
@@ -65,3 +77,88 @@ func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (Ba
 		ProvisioningDHCPRange:    provisioningDHCPRange,
 	}, nil
 }
+
+func getProvisioningIPCIDR(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningNetworkCIDR != "" && baremetalConfig.ProvisioningIp != "" {
+		_, net, err := net.ParseCIDR(baremetalConfig.ProvisioningNetworkCIDR)
+		if err == nil {
+			cidr, _ := net.Mask.Size()
+			generatedConfig := fmt.Sprintf("%s/%d", baremetalConfig.ProvisioningIp, cidr)
+			return &generatedConfig
+		}
+	}
+	return nil
+}
+
+func getDeployKernelUrl(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningIp != "" {
+		generatedConfig := fmt.Sprintf("http://%s/%s", net.JoinHostPort(baremetalConfig.ProvisioningIp, baremetalHttpPort), baremetalKernelUrlSubPath)
+		return &generatedConfig
+	}
+	return nil
+}
+
+func getDeployRamdiskUrl(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningIp != "" {
+		generatedConfig := fmt.Sprintf("http://%s/%s", net.JoinHostPort(baremetalConfig.ProvisioningIp, baremetalHttpPort), baremetalRamdiskUrlSubPath)
+		return &generatedConfig
+	}
+	return nil
+}
+
+func getIronicEndpoint(baremetalConfig BaremetalProvisioningConfig) *string {
+	if baremetalConfig.ProvisioningIp != "" {
+		generatedConfig := fmt.Sprintf("http://%s/%s", net.JoinHostPort(baremetalConfig.ProvisioningIp, baremetalIronicPort), baremetalIronicEndpointSubpath)
+		return &generatedConfig
```

Comment:
is this always http? non https?

Reply:
https://github.com/openshift/machine-api-operator/pull/478#discussion_r370268792